lfs/batcher.go

```diff
@@ -6,36 +6,43 @@ import "sync/atomic"
 // be added to the batcher from multiple goroutines and pulled off in groups
 // when one of the following conditions occurs:
 //   * The batch size is reached
+//   * Truncate() is called, forcing the batch to be returned immediately, as-is
 //   * Exit() is called
-// When an Exit() occurs, the group may be smaller than the batch size.
+// When an Exit() or Truncate() occurs, the group may be smaller than the batch
+// size.
 type Batcher struct {
 	exited     uint32
 	batchSize  int
 	input      chan interface{}
 	batchReady chan []interface{}
+	truncate   chan interface{}
 }
 
 // NewBatcher creates a Batcher with the batchSize.
 func NewBatcher(batchSize int) *Batcher {
 	b := &Batcher{
 		batchSize:  batchSize,
-		input:      make(chan interface{}, batchSize),
+		input:      make(chan interface{}),
 		batchReady: make(chan []interface{}),
+		truncate:   make(chan interface{}),
 	}
 
 	go b.acceptInput()
 	return b
 }
 
-// Add adds an item to the batcher. Add is safe to call from multiple
-// goroutines.
-func (b *Batcher) Add(t interface{}) {
+// Add adds an item (or many items) to the batcher. Add is safe to call from
```

Comment:
Way clearer! Changed in https://github.com/github/git-lfs/pull/1528/commits/49489c0f43011c5f0638eb2fd17b993e7488a0a6.